packages/node_modules/@ciscospark/plugin-memberships/src/memberships.js

```diff
@@ -24,6 +38,148 @@ import {SparkPlugin, Page} from '@ciscospark/spark-core';
  * @name Memberships
  */
 const Memberships = SparkPlugin.extend({
+  /**
+   * Connect to the web socket to listen to incoming messages.
+   * @returns {Promise}
+   */
+  listen() {
+    // Create a common envelope that we will wrap all events in
+    this.eventEnvelope = {
+      resource: 'memberships',
+      // orgId -- TODO get from this.spark.internal.user
+      // createdBy -- TODO get from this.spark.internal.user
+      ownedBy: 'creator',
+      status: 'active',
+      created: new Date().toISOString(),
+      data: {}
+    };
+    // Register to listen to events
+    return this.spark.internal.mercury.connect()
+      .then(() => this.listenTo(this.spark.internal.mercury, CONVERSATION_ACTIVITY,
+        (event) => this.onWebexApiEvent(event)));
+  },
+
+  /**
+   * Trigger a membership related events.
+   * @param {Object} event
+   * @returns {undefined}
+   */
+  onWebexApiEvent(event) {
+    const {activity} = event.data;
+    console.log(activity);
+
+    // Reply activities are not currently supported
+    if (activity.type === API_ACTIVITY_TYPE.REPLY) {
+      return;
+    }
+
+    /* eslint-disable no-case-declarations */
+    // eslint-disable-next-line padding-line-between-statements
+    switch (activity.verb) {
+      case API_ACTIVITY_VERB.ADD:
+        const membershipCreatedEventData = this.getCreatedEvent(activity);
+        debug(`membership "created" payload: ${JSON.stringify(membershipCreatedEventData)}`);
+        this.trigger(MEMBERSHIP_CREATED, membershipCreatedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.LEAVE:
+        const membershipDeletedEventData = this.getDeletedEvent(activity);
+        debug(`membership "deleted" payload: ${JSON.stringify(membershipDeletedEventData)}`);
+        this.trigger(MEMBERSHIP_DELETED, membershipDeletedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.ACKNOWLEDGE:
+      case API_ACTIVITY_VERB.ADD_MODERATOR:
+      case API_ACTIVITY_VERB.REMOVE_MODERATOR:
+        const membershipUpdatedEventData = this.getUpdatedEvent(activity);
+        debug(`membership "updated" payload: ${JSON.stringify(membershipUpdatedEventData)}`);
+        this.trigger(MEMBERSHIP_UPDATED, membershipUpdatedEventData);
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Constructs event data object for the "membership created" event,
+   * adhering to Hydra's Webehook data for an "created" event on the "membership" resource.
+   * @param {Object} activity from mercury
+   * @returns {Object} constructed event
+   */
+  getCreatedEvent(activity) {
```

Comment:
`getCreatedEvent`, `getDeletedEvent`, and `getUpdatedEvent` can be further simplified by passing a second argument to `getCommonSdkEvent`: the value you want on the `.event` property.

Reply:
Good point.

Reply:
Simplified.